usd/src/sdf_parser/Link.cc

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include "sdf/usd/sdf_parser/Link.hh"
+
+#include <string>
+
+#include <ignition/math/Pose3.hh>
+
+// TODO(ahcorde) this is to remove deprecated "warnings" in usd, these warnings
+// are reported using #pragma message so normal diagnostic flags cannot remove
+// them. This workaround requires this block to be used whenever usd is
+// included.
+#pragma push_macro ("__DEPRECATED")
+#undef __DEPRECATED
+#include <pxr/usd/sdf/path.h>
+#include <pxr/usd/usd/stage.h>
+#include <pxr/usd/usdGeom/xform.h>
+#include <pxr/usd/usdPhysics/massAPI.h>
+#include <pxr/usd/usdPhysics/rigidBodyAPI.h>
+#pragma pop_macro ("__DEPRECATED")
+
+#include "sdf/Link.hh"
+#include "sdf/usd/sdf_parser/Light.hh"
+#include "../UsdUtils.hh"
+
+namespace sdf
+{
+// Inline bracke to help doxygen filtering.
+inline namespace SDF_VERSION_NAMESPACE {
+//
+namespace usd
+{
+  UsdErrors ParseSdfLink(const sdf::Link &_link, pxr::UsdStageRefPtr &_stage,
+      const std::string &_path, bool _rigidBody)
+  {
+    const pxr::SdfPath sdfLinkPath(_path);
+    UsdErrors errors;
+
+    auto usdLinkXform = pxr::UsdGeomXform::Define(_stage, sdfLinkPath);
+
+    ignition::math::Pose3d pose;
+    auto poseErrors = sdf::usd::PoseWrtParent(_link, pose);
+    if (!poseErrors.empty())
+    {
+      for (const auto &e : poseErrors)
+        errors.push_back(e);
```

Comment:
nit: I think you can do this on one line instead of with the `for` loop

~~~
errors.insert(errors.end(), poseErrors.begin(), poseErrors.end());
~~~

Reply:
https://github.com/ignitionrobotics/sdformat/commit/16b50cbe86e6778669d26890008d1fe7d56078c5